lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -0,0 +1,12 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi < DelegateClass(ActiveModelSerializers::Adapter::JsonApi)
+        def initialize(serializer, options = {})
```

Comment:
you should be able to `def initialize(*)` to pass through all args whatever the method signature and then just call `super` after the warning